azerothcore/azerothcore-wotlk
# fix(DB/conditions): Prevent NPC multiple spawns

## Changes Proposed:
- Add Condition to check if NPC is present already.

## Issues Addressed:

- Closes  [[Display error] Sha'ni Proudtusk (npc=9136) not despawning #3106](https://github.com/chromiecraft/chromiecraft/issues/3106)

## SOURCE:


## Tests Performed:
 Had quest [Dreadmaul Rock](https://classic.wowhead.com/quest=3821/dreadmaul-rock). Clicked on gameobject. Now only spawns one NPC wait for de-spawn and click again spawns again. Relog to different character wait for de-spawn and tried again.

**She will appear in ghost form if you have the quest.** Source : [wowwiki](https://wowwiki-archive.fandom.com/wiki/Sha%27ni_Proudtusk) notes.


## How to Test the Changes:


1. Accept quest [Dreadmaul Rock](https://classic.wowhead.com/quest=3821/dreadmaul-rock)
2. Click on required gameobject couple times
3. Should only spawn one npc.
4. Wait for npc to de-spawn. Repeat steps 2-3.
5. Relog to different character. Repeat steps 1-4.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
